plugin/pkg/admission/resourcequota/controller.go

```diff
@@ -407,6 +407,20 @@ func CheckRequest(quotas []corev1.ResourceQuota, a admission.Attributes, evaluat
 	// if we have limited resources enabled for this resource, always calculate usage
 	inputObject := a.GetObject()
 
+	// if request is for delete, no need to calculate usage
+	// since admission control is incapable of guaranteeing a DELETE request actually succeeds.
+	metadata, err := meta.Accessor(inputObject)
+	if err != nil {
+		return quotas, nil
+	}
+	if metadata != nil {
+		deletionTimestamp := metadata.GetDeletionTimestamp()
```

Comment:
this is special-casing a scenario where the object is being deleted, but a more fundamental issue seems to be that we're blocking updates that don't actually change consumed quota.

rather than short-circuiting on deletionTimestamp, it seems like we should fix the issue where an update request that doesn't increase consumed quota is blocked by quota admission

Reply:
we already do that zero-delta check and short-circuit for update requests, but we do it *after* checking uncovered global quota. I think that check should move first.

Something like https://github.com/liggitt/kubernetes/commits/short-circuit-zero-delta-quota might make more sense... 

If you want to pull that commit in and make sure there is test coverage for zero-delta updates with global quota, that would be great.

Reply:
Thanks for the pointer. I'll look at it to understand and update the PR.

Reply:
Thanks. Let me know if you have any questions. I'd like to see this fixed

Reply:
I've pulled in your commit and removed the early short-circuiting on deletionTimestamp.
Looking at the existing tests, I think this will cover zero-delta update code path https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/resourcequota/admission_test.go#L438. Do you think we need a separate test for this.

Reply:
The existing tests clearly didn't cover zero delta with global quota. We need tests that demonstrate the failure this is fixing